streampipes-client-python/streampipes_client/functions/function_handler.py

```diff
@@ -0,0 +1,192 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import asyncio
+import json
+import logging
+from typing import AsyncIterator, Dict, List
+
+from streampipes_client.client.client import StreamPipesClient
+from streampipes_client.functions.broker.broker import Broker
+from streampipes_client.functions.broker.nats_broker import NatsBroker
+from streampipes_client.functions.registration import Registration
+from streampipes_client.functions.utils.async_iter_handler import AsyncIterHandler
+from streampipes_client.functions.utils.data_stream_context import DataStreamContext
+from streampipes_client.functions.utils.function_context import FunctionContext
+from streampipes_client.model.resource.data_stream import DataStream
+
+logger = logging.getLogger(__name__)
+
+
+# TODO Exception should be removed once all brokers are implemented.
+class UnsupportedBroker(Exception):
+    """Exception if a broker isn't implemented yet."""
+
+    def __init__(self, message):
+        super().__init__(message)
+
+
+class FunctionHandler:
+    """The function handler manages the StreamPipesFunctions.
+    It controls the connection to the brokers, starts the functions, manages the broadcast of the live data
+    and is able to stop the connection to the brokers and functions.
+
+    Parameters
+    ----------
+    registration: Registration
+        The registration, that contains the StreamPipesFunctions.
+    client: StreamPipesClient
+        The client to interact with the API.
+    """
+
+    def __init__(self, registration: Registration, client: StreamPipesClient) -> None:
+        self.registration = registration
+        self.client = client
```

Comment:
I think this kind of breaks the basic idea of how we want to structure the API of the Python client.
But that's not really important for now, I'll give it some thought and come up with a solution.